### DISCUSSION TITLE
forge build --zksync

### DISCUSSION THREAD

**Discussion by rrocode** [Cyfrin/foundry-full-course-cu#2820] [Q&A]
Hello,

While trying to build using 
```
foundry build --zksync
```
I get following error in terminal:
```
[⠊] Using zksolc-1.5.4
[⠒] Compiling (zksync)
Error: 
solc exited with signal: 6 (SIGABRT)
dyld: Symbol not found: __ZTTNSt3__114basic_ifstreamIcNS_11char_traitsIcEEEE
  Referenced from: /Users/MyUser/.zksync/solc-macosx-amd64-0.8.28-1.0.1 (which was built for Mac OS X 12.7)
  Expected in: /usr/lib/libc++.1.dylib
 in /Users/MyUser/.zksync/solc-macosx-amd64-0.8.28-1.0.1
```

foundry zksync is installed and update of forge and cast is through using below command:
```
foundryup-zksync
```

forge version is
```
forge 0.0.2 (d9babce 2024-11-06T00:34:25.136801000Z)
```

I've tried asking ChatGPT and here what is says:
```
The error you're seeing likely relates to a compatibility issue between your macOS version and the zksync Solidity compiler (zksolc) binary. The error dyld: Symbol not found: __ZTTNSt3__114basic_ifstream... often occurs when binaries are compiled for different macOS versions or libraries than what is available on your system.

To resolve this, try the following steps:

Upgrade macOS: Ensure your macOS version is compatible with the zksync compiler.
Reinstall zksolc: Try removing and reinstalling zksolc, or check for a compatible release.
Custom Compiler Path: Configure the path in foundry.toml to specify a compatible version.
For further details, you can refer to the [Foundry-ZKsync documentation](https://foundry-book.zksync.io/).
``` 

Did anyone had similar issues? Does this mean that i can't run foundry with --zksync on my OS Version?

Any help is very appreciated, thanks!

**Comment by EngrPips** [comment]
Hello @rrocode, read this https://github.com/Cyfrin/foundry-full-course-cu/discussions/2466 discussion. It may help you.

**Comment by rrocode** [comment]
Just an addition, switching back to vanilla foundry with
```
foundryup
```
and building with 
```
forge build
```
results in
```
[⠒] Compiling...
[⠢] Compiling 1 files with Solc 0.8.25
[⠆] Solc 0.8.25 finished in 3.02s
Compiler run successful!
```

**Reply by EngrPips** [reply]
You had an issue with `foundry-zksync` compiler, not foundry compiler, so this makes sense.

**Reply by rrocode** [reply]
no, not really, didn't help. Will investigate bit more, don't want to upgrade OS for this only reason. Thanks!

**Reply by EngrPips** [reply]
Oh I see, I am looking forward to hearing how things play out.

**Comment by rrocode** [comment]
Ok, after some more investigation, it seems that most of google resources on 
```
Expected in: /usr/lib/libc++.1.dylib
```
have to do with OS version. In my case, it's lower that 12 as stated in 
```
Referenced from: /Users/MyUser/.zksync/solc-macosx-amd64-0.8.28-1.0.1 (which was built for Mac OS X 12.7)
```
While i currently don't want to upgrade neither have time for it, will have to first live without zksync for a while. **But that would be my next step - to upgrade the MacOS.**

Interestingly, you can as well run the build command under specific zk compiler, but this didn't help:
```
forge build --zk-solc-path /Users/MyUser/.zksync/solc-macosx-amd64-0.8.28-1.0.1 --zksync
forge build --zk-solc-path /Users/MyUser/.zksync/solc-macosx-amd64-0.8.18-1.0.1 --zksync
```

i'll come back to this issue once i have some more time and through the OS upgrade.

**Reply by EngrPips** [reply]
Interesting, The `ZKSync` part of the course is actually optional and you can come back to it later.

**Reply by mustby** [reply]
Hey @EngrPips, any idea on this?

I'm running: foundryup-zksync

And I get this error: bash: foundryup-zksync: command not found

**Reply by EngrPips** [reply]
Hey @mustby, Have you initially installed `foundry-zksync`?

**Reply by mustby** [reply]
Hey! No I haven't. But I just tried running that and I got the same error message:

bash: foundry-zksync: command not found

**Reply by EngrPips** [reply]
Please follow the instructions on this page https://foundry-book.zksync.io/getting-started/installation to install `FoundryZKSync`

**Reply by mustby** [reply]
working now! Thank you

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out. Keep crushing it.
